src/state.rs

```diff
@@ -210,14 +210,11 @@ impl DkgState {
     fn parts_into_acks<R: bls::rand::RngCore>(
         &mut self,
         parts: BTreeSet<IdPart>,
-        mut rng: R,
+        rng: &mut R,
     ) -> Result<DkgVote> {
         let mut acks = BTreeMap::new();
         for (sender_id, part) in parts {
-            match self
-                .keygen
-                .handle_part(&sender_id, part.clone(), &mut rng)?
-            {
+            match self.keygen.handle_part(&sender_id, part.clone(), rng)? {
```

Comment:
It would be better to update `keygen.handle_part(..)` to take `mut rng: R` instead of forcing the caller to pass a mutable reference.